test/jdk/java/awt/Dialog/FileDialogIconTest/FileDialogIconTest.java

```diff
@@ -169,90 +182,103 @@ public void actionPerformed(ActionEvent event) {
             }
         });
 
         setImageButton6.addActionListener(new ActionListener() {
             public void actionPerformed(ActionEvent event) {
-                List<BufferedImage> list = new ArrayList<>();
+                List<Image> list = new ArrayList<>();
                 try {
                     Robot robot = new Robot();
                     Rectangle rectangle;
                     for (int i = 1; i <= 4; i++) {
-                        rectangle = new Rectangle(i * 10, i * 10, i * 10 + 40, i * 10 + 40);
-                        java.awt.image.BufferedImage image = robot.createScreenCapture(rectangle);
+                        rectangle = new Rectangle(i * 10, i * 10,
+                                                  i * 10 + 40, i * 10 + 40);
+                        BufferedImage image = robot.createScreenCapture(rectangle);
                         robot.delay(100);
                         list.add(image);
                     }
+                    setImagesToFD(list);
                 } catch (Throwable t) {
                     t.printStackTrace();
                 }
-                PassFailJFrame.log("Captured images and set to the list for frame");
+                PassFailJFrame.log("Captured images and set to the list for dialog");
             }
         });
 
         Button buttonLoad = new Button("Load Dialog");
         Button buttonSave = new Button("Save Dialog");
-        Button buttonSimple = new Button("Just Dialog");
+        Button buttonSimple = new Button("Simple Dialog");
         buttonLoad.addActionListener(new MyActionListener(FileDialog.LOAD, "LOAD"));
         buttonSave.addActionListener(new MyActionListener(FileDialog.SAVE, "SAVE"));
         buttonSimple.addActionListener(new MyActionListener(-1, ""));
 
-        frame.setSize(400, 400);
-        frame.setLayout(new FlowLayout());
-        frame.add(buttonLoad);
-        frame.add(buttonSave);
-        frame.add(buttonSimple);
-        frame.add(setImageButton1);
-        frame.add(setImageButton2);
-        frame.add(setImageButton3);
-        frame.add(setImageButton4);
+        frame.setLayout(new GridLayout(2, 1));
+
+        Panel panel1 = new Panel(new GridLayout(1, 3));
+        panel1.add(buttonLoad);
+        panel1.add(buttonSave);
+        panel1.add(buttonSimple);
+
+        Panel panel2 = new Panel(new GridLayout(3, 2));
+        panel2.add(setImageButton1);
+        panel2.add(setImageButton2);
+        panel2.add(setImageButton3);
+        panel2.add(setImageButton4);
+        panel2.add(setImageButton5);
+        panel2.add(setImageButton6);
+
+        frame.add(panel1);
+        frame.add(panel2);
         frame.pack();
+
         return frame;
     }
-}
 
-class MyActionListener implements ActionListener {
-    int id;
-    String name;
+    static class MyActionListener implements ActionListener {
+        int id;
+        String name;
```

Comment:
```suggestion
        final int id;
        final String name;
```
If the value of the fields is unchanged, they should be `final`. The may even be `private`.